api/src/services/script-runner.service.ts

```diff
@@ -1335,13 +1335,15 @@ export class ScriptRunnerService {
       .concat(householdMembers)
       .map((address) => address.workAddressId);
 
-    await this.prisma.address.deleteMany({
-      where: {
-        id: {
-          in: workAddressesIds,
+    for (let i = 0; i < workAddressesIds.length; i += 10000) {
```

Comment:
**issue:** I wrote this up in a DM but posting here for visibility too. I think you should change it so that each iteration of the for loop does:
- reads in 10k applications / household members
- updates 10k application/household members
- deletes 10k addresses